File: agent_misc/mcps/codeql-development-mcp-server/server/ql/python/tools/test/PrintCFG/Example1.py
# Example1 test case for PrintCFG.ql

# COMPLIANT - Basic control flow for CFG testing
def simple_method(x):
    if x > 0:
        print("Positive")
    else:
        print("Non-positive")

    for i in range(3):
        print(i)

# NON_COMPLIANT - Function with complex control flow
def complex_method(value):
    if value < 0:
        return -1

    while value > 10:
        value = value // 2

    if value == 0:
        return 0
    elif value == 1:
        return 1
    else:
        return value * 2

File: agent_misc/mcps/codeql-development-mcp-server/server/ql/csharp/tools/test/PrintCFG/Example1.cs
// Example1 test case for PrintCFG.ql
using System;

public class Example1
{
    public static void Main(string[] args)
    {
        // COMPLIANT - Basic control flow for CFG testing
        int x = 5;

        if (x > 0)
        {
            Console.WriteLine("Positive");
        }
        else
        {
            Console.WriteLine("Non-positive");
        }

        for (int i = 0; i < 3; i++)
        {
            Console.WriteLine(i);
        }
    }

    // NON_COMPLIANT - Method with complex control flow
    public static int ComplexMethod(int value)
    {
        if (value < 0)
        {
            return -1;
        }

        while (value > 10)
        {
            value = value / 2;
        }

        switch (value)
        {
            case 0:
                return 0;
            case 1:
                return 1;
            default:
                return value * 2;
        }
    }
}
